src/redfishpower/redfishpower.c

```diff
@@ -306,16 +336,43 @@ static void powermsg_destroy(struct powermsg *pm)
     }
 }
 
+static struct plug_data *plug_data_create(const char *hostname)
+{
+    struct plug_data *pd = calloc(1, sizeof(*pd));
+    if (!pd)
+        err_exit(true, "calloc");
+    pd->hostname = xstrdup(hostname);
+    return pd;
+}
+
+static void plug_data_destroy(struct plug_data *pd)
+{
+    if (pd) {
+        xfree(pd->hostname);
+        free(pd);
+    }
+}
+
 static struct powermsg *stat_cmd_host(CURLM * mh, char *hostname)
 {
-    struct powermsg *pm = powermsg_create(mh,
-                                          hostname,
-                                          CMD_STAT,
-                                          statpath,
-                                          NULL,
-                                          NULL,
-                                          0,
-                                          STATE_SEND_POWERCMD);
+    struct powermsg *pm;
+    char *target_hostname;
+    char *plugname;
+
+    if (get_host_and_plug(hostname, &target_hostname, &plugname) < 0) {
+        printf("unknown host/plug specified: %s\n", hostname);
+        return NULL;
+    }
```

Comment:
Suggestion: rename the `hostname` parameter of `stat_cmd_host()` to `plug`.  Maybe "host" shouldn't be in the function name either.

Error message should say "unknown plug specified".

The thing powerman wants you to operate on is always a plug, even if it happens to be the same as a hostname.